examples/gosherve-demo/pyproject.toml

```diff
@@ -0,0 +1,146 @@
+[project]
+name = "gosherve-k8s-operator"
+version = "2.0.0"
+requires-python = ">=3.12"
+authors = [{name="Jon Seager"}]
+classifiers = [
+    "Programming Language :: Python :: 3",
+    "License :: OSI Approved :: Apache Software License",
+    "Development Status :: 5 - Production/Stable",
+    "Intended Audience :: Developers",
+    "Intended Audience :: System Administrators",
+    "Operating System :: MacOS :: MacOS X",
+    "Operating System :: POSIX :: Linux",
+]
+dependencies = [
+    "ops~=3.0",
+]
+
+[dependency-groups]
+lint = [
+    "ruff==0.11.2",
+    "codespell==2.4.1",
+]
+static = [
+    "ops[testing,tracing]",
+    "pyright==1.1.385",
+]
+unit = [
+    "ops[testing,tracing]",
+    "pytest~=8.0",
+    "coverage[toml]~=7.0",
+]
+integration = [
+    "ops[testing,tracing]",
+    "pytest~=8.0",
+    "jubilant~=1.2",
+]
+
+# Testing tools configuration
+[tool.coverage.run]
+branch = true
+
+[tool.coverage.report]
+show_missing = true
+
+# Formatting tools configuration
+[tool.autopep8]
+max-line-length = 99
+ignore = ["W503"]
+recursive = true
+jobs = -1
+aggressive = 3
+
+# Linting tools configuration
+[tool.ruff]
+line-length = 99
+target-version = "py312"
+
+# Ruff formatter configuration
+[tool.ruff.format]
+quote-style = "single"
+exclude = [
+    "lib/*",  # Ignore libs fetched during our testing (or by a charmer)
+]
+
+[tool.ruff.lint]
```

Comment:
WDYT about not specifying tool config here and relying on monorepo config instead?